test/transform_tests.erl

```diff
@@ -136,11 +141,30 @@ remote_type_refl_test() ->
 
 %% -----------------------------------------------------------------------------
 
-%% TODO: Test special attributes
-
 -type ignored() :: string().
 
--type url() :: string().
+-type uri() :: string().
+
+verify_uri(Str) ->
+  is_map(uri_string:parse(Str)).
+
+verify_test() ->
+  {?type_refl, #{check := Check}} = uri(),
+  ?assertEqual(fun ?MODULE:verify_uri/1, Check).
+
+%% -----------------------------------------------------------------------------
+
+-type ipv4_address() :: {byte(), byte(), byte(), byte()}.
+
+parse_ip_address(Str) ->
+  {ok, Addr} = inet:parse_ipv4_address(Str),
```

Comment:
I don't remember why I use throwing version of the function internally. I think this should be changed to `{ok, Val} | {error, Term}`